dbms/src/Common/XDBCBridgeHelper.h

```diff
@@ -0,0 +1,293 @@
+#include <Poco/File.h>
+#include <Poco/Net/HTTPRequest.h>
+#include <Poco/Path.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Common/ShellCommand.h>
+#include <Common/config.h>
+#include <common/logger_useful.h>
+#include <ext/range.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int EXTERNAL_EXECUTABLE_NOT_FOUND;
+    extern const int EXTERNAL_SERVER_IS_NOT_RESPONDING;
+}
+
+/**
+ * Class for Helpers for Xdbc-bridges, provide utility methods, not main request
+ */
+class IXDBCBridgeHelper
+{
+public:
+    static constexpr inline auto DEFAULT_FORMAT = "RowBinary";
+
+    virtual std::vector<std::pair<std::string, std::string>> getURLParams(const std::string & cols, size_t max_block_size) const = 0;
+    virtual void startBridgeSync() const = 0;
+    virtual Poco::URI getMainURI() const = 0;
+    virtual Poco::URI getColumnsInfoURI() const = 0;
+    virtual IdentifierQuotingStyle getIdentifierQuotingStyle() = 0;
+    virtual String getName() const = 0;
+
+    virtual ~IXDBCBridgeHelper() {}
+};
+
+using BridgeHelperPtr = std::shared_ptr<IXDBCBridgeHelper>;
+
+template <typename BridgeHelperMixin>
+class XDBCBridgeHelper : public IXDBCBridgeHelper
+{
+private:
+    Poco::Timespan http_timeout;
+
+    std::string connection_string;
+
+    Poco::URI ping_url;
+
+    Poco::Logger * log = &Poco::Logger::get(BridgeHelperMixin::getName() + "BridgeHelper");
+
+    std::optional<IdentifierQuotingStyle> quote_style;
+
+protected:
+    auto getConnectionString() const
+    {
+        return connection_string;
+    }
+
+public:
+
+    using Configuration = Poco::Util::AbstractConfiguration;
+
+    const Configuration & config;
+
+    static constexpr inline auto DEFAULT_HOST = "localhost";
+    static constexpr inline auto DEFAULT_PORT = BridgeHelperMixin::DEFAULT_PORT;
+    static constexpr inline auto PING_HANDLER = "/ping";
+    static constexpr inline auto MAIN_HANDLER = "/";
+    static constexpr inline auto COL_INFO_HANDLER = "/columns_info";
+    static constexpr inline auto IDENTIFIER_QUOTE_HANDLER = "/identifier_quote";
+    static constexpr inline auto PING_OK_ANSWER = "Ok.";
+
+    XDBCBridgeHelper(
+            const Configuration & config_, const Poco::Timespan & http_timeout_, const std::string & connection_string_)
+            : http_timeout(http_timeout_), connection_string(connection_string_), config(config_)
+    {
+        size_t bridge_port = config.getUInt(BridgeHelperMixin::configPrefix() + ".port", DEFAULT_PORT);
+        std::string bridge_host = config.getString(BridgeHelperMixin::configPrefix() + ".host", DEFAULT_HOST);
+
+        ping_url.setHost(bridge_host);
+        ping_url.setPort(bridge_port);
+        ping_url.setScheme("http");
+        ping_url.setPath(PING_HANDLER);
+    }
+
+    virtual ~XDBCBridgeHelper() {}
+
+    String getName() const override
+    {
+        return BridgeHelperMixin::getName();
+    }
+
+    IdentifierQuotingStyle getIdentifierQuotingStyle() override
```

Comment:
Maybe startBridgeSync() should be called here?

Reply:
Done.